code_samples/fabric_kusto/fabric_kusto_vector_query.ipynb

```diff
@@ -0,0 +1,469 @@
+    "kdb_authority_id = os.getenv(\"KUSTO_AUTHORITY_ID\")\n",
+    "if kdb_authority_id is None or kdb_authority_id == \"\":\n",
+    "    print(\"KUSTO_AUTHORITY_ID environment variable not set.\")\n",
+    "    exit()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "from openai import AzureOpenAI\n",
+    "from PIL import Image\n",
+    "\n",
+    "from azure.kusto.data import KustoClient, KustoConnectionStringBuilder\n",
+    "from azure.kusto.data.helpers import dataframe_from_result_table\n",
+    "\n",
+    "\n",
+    "cluster = f\"https://{kdb_cluster_name}.kusto.fabric.microsoft.com/\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### [Option 1] AAD Device Authentication"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kcsb = KustoConnectionStringBuilder.with_aad_device_authentication(cluster)\n",
+    "client = KustoClient(kcsb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### [Option 2] Client-Secret Authentication \n",
+    "To use client-secret authentication, you need to set the following environment variables in the `.env` file:\n",
+    "```bash\n",
+    "KUSTO_CLIENT_ID=<client id of the service principle>\n",
+    "KUSTO_CLIENT_SECRET=<client secret of the service principle>\n",
+    "KUSTO_AUTHORITY_ID=<authority id>\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kcsb = KustoConnectionStringBuilder.with_aad_application_key_authentication(cluster, kdb_client_id, kdb_client_secret, kdb_authority_id)\n",
+    "client = KustoClient(kcsb)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Helper methods"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def search_kusto(\n",
+    "    kusto_client: KustoClient,\n",
+    "    kusto_db: str,\n",
+    "    query_vector: list,\n",
+    "    index_name: str,\n",
+    "    vector_fields: list, \n",
+    "    key_field: str = \"id\",\n",
+    "    return_fields: list = [],\n",
+    "    k: int = 20,\n",
+    "    filter: str = ''\n",
+    "):\n",
+    "    if not vector_fields: \n",
+    "        raise(\"vector_field is required\")\n",
+    "    \n",
+    "    queries = []\n",
+    "    for v_field in vector_fields:\n",
+    "        queries.append(f\"\"\"\n",
+    "        let {v_field}_view = view() {{ \n",
+    "            ['{index_name}'] \n",
+    "            | summarize arg_max(ingestion_time(), *) by ['{key_field}'] \n",
+    "            | extend similarity=series_cosine_similarity_fl('{query_vector}', ['{v_field}'], 1, 1)\n",
```

Comment:
We should either:

1) add some comment above this function that the user needs to create this function

https://learn.microsoft.com/en-us/azure/data-explorer/kusto/functions-library/series-cosine-similarity-fl?tabs=stored#function-definition

or 
2) add a step to this Python function that creates the stored function if it doesn't exist

Reply:
Hey @jose-perales , thanks for the comments! 
I changed the `series_cosine_similarity_fl` back to use the native `series_cosine_similarity`.  Also added comment above for `series_cosine_similarity` that user can check more details about it following the attached link.